core/chainio/avs_writer.go

```diff
@@ -166,18 +153,152 @@ func (w *AvsWriter) compareAggregatorBalance(amount *big.Int, aggregatorAddress
+		return nil
 	}
 	w.logger.Info("Batcher balance", "balance", batcherBalance)
 	if batcherBalance.Cmp(amount) < 0 {
 		return fmt.Errorf("cost is higher than Batcher balance")
 	}
 	return nil
 }
+
+// |---RETRYABLE---|
+
+func (w *AvsWriter) RespondToTaskV2Retryable(opts *bind.TransactOpts, batchMerkleRoot [32]byte, senderAddress common.Address, nonSignerStakesAndSignature servicemanager.IBLSSignatureCheckerNonSignerStakesAndSignature) (*types.Transaction, error) {
+	var (
+		tx  *types.Transaction
+		err error
+	)
+	respondToTaskV2_func := func() (*types.Transaction, error) {
+		tx, err = w.AvsContractBindings.ServiceManager.RespondToTaskV2(opts, batchMerkleRoot, senderAddress, nonSignerStakesAndSignature)
+		if err != nil {
+			tx, err = w.AvsContractBindings.ServiceManagerFallback.RespondToTaskV2(opts, batchMerkleRoot, senderAddress, nonSignerStakesAndSignature)
+			if err != nil {
+				// Note return type will be nil
+				if err.Error() == "not found" {
+					err = connection.TransientError{Inner: err}
+					return tx, err
+				}
+				if strings.Contains(err.Error(), "connect: connection refused") {
+					err = connection.TransientError{Inner: err}
+					return tx, err
+				}
+				if strings.Contains(err.Error(), "read: connection reset by peer") {
+					return tx, connection.TransientError{Inner: err}
+				}
+				err = connection.PermanentError{Inner: fmt.Errorf("Permanent error: Unexpected Error while retrying: %s\n", err)}
+			}
+		}
+		return tx, err
+	}
+	return connection.RetryWithData(respondToTaskV2_func, connection.MinDelay, connection.RetryFactor, connection.NumRetries)
+}
+
+func (w *AvsWriter) BatchesStateRetryable(arg0 [32]byte) (struct {
+	TaskCreatedBlock      uint32
+	Responded             bool
+	RespondToTaskFeeLimit *big.Int
+}, error) {
+	var (
+		state struct {
+			TaskCreatedBlock      uint32
+			Responded             bool
+			RespondToTaskFeeLimit *big.Int
+		}
+		err error
+	)
+	batchesState_func := func() (struct {
+		TaskCreatedBlock      uint32
+		Responded             bool
+		RespondToTaskFeeLimit *big.Int
+	}, error) {
+		state, err = w.AvsContractBindings.ServiceManager.BatchesState(&bind.CallOpts{}, arg0)
+		if err != nil {
+			state, err = w.AvsContractBindings.ServiceManagerFallback.BatchesState(&bind.CallOpts{}, arg0)
+			// If error is not nil throw out result in state!
+			if err != nil {
+				// Note return type will be nil
+				if err.Error() == "not found" {
+					err = connection.TransientError{Inner: err}
+					return state, err
+				}
+				if strings.Contains(err.Error(), "connect: connection refused") {
+					err = connection.TransientError{Inner: err}
+					return state, err
+				}
+				if strings.Contains(err.Error(), "read: connection reset by peer") {
+					return state, connection.TransientError{Inner: err}
+				}
+				err = connection.PermanentError{Inner: fmt.Errorf("Permanent error: Unexpected Error while retrying: %s\n", err)}
+			}
```

Comment:
Maybe an auxiliary function just to handle the received errors might work

Reply:
Let's not try to abstract this, since it may not be exactly the same always.

But use a case statemente, and group errors of the same type in the same cas:

```if err != nil {
    switch {
    case strings.Contains(err.Error(), "not found"),
         strings.Contains(err.Error(), "connect: connection refused"),
         strings.Contains(err.Error(), "read: connection reset by peer"):
        err = connection.TransientError{Inner: err}
    default:
        err = connection.PermanentError{Inner: fmt.Errorf("Permanent error: Unexpected Error while retrying: %v", err)}
    }
    return state, err
}```

Reply:
I elected to just remove these checks and add more explicit permanent errors in a later pr